### PR TITLE
Story CORE-462 | Update old docs

### PR DESCRIPTION
**Pull request by franciscobonand**
# Description

Kind: Story

Section: Documentation

Summary: Update outdated docs

## Changelog

> Updated most of the files in `/docs` folder
> Fixed Ping Pong example
> Fixed resolved channel env var injection in `converter.go` so it injects the original channel to reference the resolved one
> Updated scripts that build and install Inspr CLI
> Updated release cretion GitHub Action so it pushes the CLI once it's build
> Renamed folder from `apply_channel_type.go` to `apply_type.go`
> Added/updated some logs, error messages and method comments all around

## Pay attention to

> Grammar errors

### CODE REVIEW

File: docs/dapp_overview.md [hunk 1]
```diff
@@ -22,54 +23,68 @@ The unique characteristics of a Node are defined by three structures set by the
...
-Channels are the structures that allow dApps and Nodes to exchange information between each other.  
+Channels are the structures that allow dApps and Nodes to exchange information between each other via message brokers.  
 A dApp/Node inputs and outputs are defined by it's Boundary. The Boundary, on the other hand, references a parent dApp's Channel (or Alias, which will be described ahead).
 
 ![channel](img/chan-func.png)  
 
-Channels are created to receive and send a specific kind of information. This specification is defined by a **Channel's Type**, which is a reference to a Type structure defined in the same context (in the same dApp) as the Channel. In other words:
-- A Channel only carries a specific kind of data (such as *int*, *string*, or an object defined by the user).
-- A Channel can't exist without a Type.  
+Channels are created to receive and send a specific kind of information. This specification is defined by it's **Type**, which is a reference to a Type structure defined in the same context (in the same dApp) as the Channel. In other words:
```
Comment (nicholaspcr) [inline docs/dapp_overview.md]: its `type`, no?

**Review by nicholaspcr — APPROVED** [review]

File: docs/dapp_overview.md (hunk 1 above)
Comment (franciscobonand) [reply docs/dapp_overview.md]: Didn't really understand your comment :confused:

Comment (franciscobonand) [reply docs/dapp_overview.md]: Yes, it is!

**Review by luiz-couto — APPROVED** [review]

File: docs/helm_installation.md [hunk 2]
```diff
@@ -23,26 +23,40 @@ $ helm upgrade --install insprd inspr/insprd
...
+| sidecarClient.writePort | Port which the Load Balancer Sidecar will receive write requests from the Sidecar Client | 3048 |
+| lbSidecar.port | Port which the Load Balancer Sidecar will receive read requests and redirect to the Sidecar Client | 3051 |
+| lbSidecar.readPort | Port which the Load Balancer Sidecar is running | 3047 |
+| lbsidecarImage | Load Balancer sidecar image | gcr.io/red-inspr/inspr/sidecar/lbsidecar |
 | insprAppsNamespace | Kubernetes namespace on which Inspr apps will be instantiated | default |
 
 ## Exposing Inspr via NGINX
 
-Inspr Controller will be exposed via NGINX by default, so you just need to have NGINX installed. Follow the instructions in the [official NGINX documentation](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm/) to install it.
+Insprd will be exposed via NGINX by default, so you just need to have NGINX installed. Follow the instructions in the [official NGINX documentation](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm/) to install it.
```
Comment (nicholaspcr) [inline docs/helm_installation.md]: Recently the nginx stable repository has been deprecated, there is a new repo that contains the updated charts of the nginx https://kubernetes.github.io/ingress-nginx/deploy/

Comment (franciscobonand) [reply docs/helm_installation.md]: Ok, will update it

**Review by nicholaspcr — APPROVED** [review]

File: docs/workspace_init.md [hunk 3]
```diff
@@ -373,39 +305,35 @@ build:
...
-
+And to execute the Makefile through the terminal you just run:
 ```zsh
 make
 ```
 
 ## Creating Inspr Workspace
 
-Now that we have our applications implemented and their Docker images available in the cluster, we're good to go and build Inspr structures to use what we created.
+Now that we have our applications implemented and their Docker images available in the cluster, we're good to go and build the Inspr structures to run the applications we created.
```
Comment (LucasStarlingdePaulaSalles) [inline docs/workspace_init.md]: good to go and build sounds weird, good to build

File: docs/workspace_init.md [hunk 4]
```diff
@@ -373,39 +305,35 @@ build:
...
 ### YAML Files
 
 First of all, from "/pingpong_demo" we access the folder "/yamls" created previously.
-
 ```zsh
 cd yamls
 ```
 
-And within this folder we will create the _.yaml_ files which describe each of the Inspr structures that will be built inside the cluster. This part of the tutorial wont take a closer look at every minimum detail on how to write the YAML files, but you can find more information about it [here](yamls/index.md).
+And within this folder we will create the `.yaml` files which describe each of the Inspr structures that will be built inside the cluster. This part of the tutorial wont take a closer look at every minimum detail on how to write the YAML files, but you can find more information about it [here](yamls/index.md).
```
Comment (LucasStarlingdePaulaSalles) [inline docs/workspace_init.md]: won't go into too much detail about how  ...

File: docs/workspace_init.md [hunk 5]
```diff
@@ -421,24 +349,24 @@ spec:
...
 
 The next file to be created is _ping.app.yaml_, in which we describe the Node that contains Ping. For better organization, let's create a new folder inside of "/yamls" called "nodes", inside of which we will store the Nodes YAMLs:
-
 ```zsh
 mkdir nodes
 touch nodes/ping.app.yaml
 ```
 
-Then, insite of _ping.app.yaml_ we must specify the kind, apiVersion and then the Node information (such as name, boundaries, image, etc.).  
+Then, insite of `ping.app.yaml` we must specify the kind, apiVersion and then the Node information (such as name, boundaries, image, etc.).
```
Comment (LucasStarlingdePaulaSalles) [inline docs/workspace_init.md]: inside?

File: docs/workspace_init.md [hunk 6]
```diff
@@ -488,139 +414,201 @@ spec:
...
   name: pptype1
-schema: yamls/types/schema.avsc
+schema: yamls/ctypes/schema.avsc
 ```
 
-Notice that the `schema` field is actually a reference to an **Avro Schema** file. By defining it like this, when a Type is created Inspr searches for the file and injects its value into the `schema` fiel. You can find more information on how schemas should be created to be used in Inspr [here](schemas_and_types.md).  
-To make everythink work properly, let's create _schema.avsc_:
-
+Notice that the `schema` field is actually a reference to an **Avro Schema** file. By defining it like this, when a Type is created Inspr searches for the file and injects its value into the `schema` field. You can find more information on how schemas should be created to be used in Inspr [here](schemas_and_types.md).  
+To make everythink work properly, let's create `schema.avsc`:
```
Comment (LucasStarlingdePaulaSalles) [inline docs/workspace_init.md]: everything

File: docs/workspace_init.md [hunk 7]
```diff
@@ -488,139 +414,201 @@ spec:
...
+```
+Invalid config file! Did you run insprctl init?
+Error: open /home/<user>/.inspr/config: no such file or directory
+```
+
+Once this is done we need to install our **UID Provider**, so we're able to create users that have access to the workspace and can manipulate it's structures.  
+
+### UID Provider installation
+
+First of all, we must retrieve the **root token for authentication**. This is the [JWT](https://jwt.io/) token that will be used to establish the first connection between Insprd and the UID Provider, generating an Admin user that will be able to create new users, as well as to manipulate Inspr's structures inside Insprd.
```
Comment (LucasStarlingdePaulaSalles) [inline docs/workspace_init.md]: admin user which will...

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Pull request merged**